boucle.py

```diff
@@ -77,17 +89,21 @@ def my_callback(frames):
                 # is there enough audio data ?
                 if clip_offset < song.length(clip):
                     length = min(song.length(clip) - clip_offset, frames)
-                    outL_buffer[:length] += song.get_data(clip,
```

Comment:
here, you can retrieve output port with clip.output value (old clip.route_out value) and `output_ports_by_name` dict (see example)